src/core/env/src/main/scala/NativeLoader.java

```diff
@@ -116,6 +116,13 @@ private void extractNativeLibraries() throws IOException{
         extractionDone = true;
     }
 
+    private void extractNativeLibraries(String libName) throws IOException{
```

Comment:
This code is not thread-safe. Is there a way you can call it inside a static block or use a lock? In Java, you can do this inside a ``synchronized`` block.